Updates for easy configuration for use in other wings

derrell opened the pull request:
This PR is primarily focused on moving all configuration into two files: `config.json` (example provided at `config.json.example`) and `credentials.json` (example provided at `credentials.json.example`). All of the individual apps' configuration files then pull the relevant values from those two files. This allows a new installation's newbie administrators to see quickly and easily what all of the configurable values are for all apps.

Because other wings use different email address formats than what NHWG uses, the format of automatically generated email addresses is now configurable in `config.json`, as configuration variable `makeMemberEmailAddress`. This variable contains the algorithm for creating the email address, which has access to the `Member` record.

This PR is provided as a DRAFT as not all changes have been tested yet.

derrell committed:
Coalesce configuration to two files: config.json and credentials.json

There were previously numerous configuration files, many of which
defined the same or similar values. There were also hard-coded values,
often uses of "NHWG" in some form or another, that should have been
configurable. This commit moves all configuration into two JSON files.
The file `config.json` contains all MIMS configuration except for
database credentials. The file `credentials.json` contains those
values.

To assist in gaining access to the JSON configuration values by
scripts, new files `getConfigValue` (for shell script access to JSON)
and `getConfig.js` (for database JavaScript access) are provided.

A new configuration variable, `makeMemberEmailAddress`, allows
configuring the expression for generating an email address of a new
member. This avoids the need for altering code when this package used
used by wings with different email address formats.

derrell committed:
wip debugging new installation and config changes

derrell committed:
wip debugging new installation and config changes

derrell committed:
fix merge conflicts

derrell committed:
add empty file mailing-list-exclude.pat needed by Mailing-List-Gen

derrell committed:
fix merge errors

derrell committed:
wip debugging new installation and config changes

derrell committed:
wip debugging new installation and config changes

derrell committed:
wip debugging new installation and config changes

derrell committed:
wip debugging new installation and config changes

derrell committed:
wip debugging new installation and config changes

derrell committed:
wip debugging new installation and config changes

derrell committed:
Merge branch 'master' of github.com:ifrguy/NHWG-MIMS

derrell committed:
wip debugging new installation and config changes

derrell committed:
wip debugging new installation and config changes

derrell committed:
move UPFIELDS and KEYS out of config file (they're for internal use)

derrell committed:
minor fixes (correct deprecations; OU change from '/'

derrell committed:
add script to generate gam commands to create OUs

derrell committed:
add new util addUserByCAPID.js; misc fixes

derrell committed:
Added AEM report showing nearest squadrons to each AEM

derrell committed:
Beginning work on Groups

derrell committed:
Reimplemented group management with initial working class AE

ifrguy commented:
All of the files in the conflicts were moved to the "MIMS" sub directory during the reorganization.

derrell committed:
Added Seniors and Cadets groups; clean up

derrell committed:
Use Members.Moderates multivalued schema property to enable group managers

derrell committed:
fix removal of moderator

derrell committed:
Added more groups

derrell committed:
Add groups Commanders and IC

derrell committed:
add groups Commanders and IC

derrell committed:
Added Directors, Logistics, Members, MissionStaff, PA, Parents, Pilots, Safety groups

derrell committed:
update group creation to include all groups

derrell committed:
uncomment region that had been temporarily commented out

derrell committed:
fix Parents group

derrell committed:
multiple fixes and additions

derrell committed:
add Professional Development group

derrell committed:
Merge remote-tracking branch 'upstream/master' into pre-merge-upstream

derrell committed:
add Holds and GroupHolds collection to seeding the database

derrell committed:
move database seed utility into INITIAL_SETUP

derrell committed:
Merge branch 'master' into pre-merge-upstream

derrell committed:
correct brokeness due to changes from upstream

derrell committed:
Merge remote-tracking branch 'upstream/master' into pre-merge-upstream

derrell committed:
Changes required by upstream file reorganization

derrell committed:
More changes required due to reorg

derrell committed:
create the public-announce group

derrell committed:
Replace Expired.py with ExpiredV2.py

The new ExpiredV2.py searches for Google accounts for which there is
no Member info, or where the Member info indicates that the member is
expired. This catches accounts for which the expiration was long
enough ago that the Member record no longer exists.

derrell committed:
update history

derrell committed:
ignore ROLE type members when suspending accounts

derrell committed:
fix typo that showed up as a result of the directory reorganization

derrell committed:
update example config

ifrguy commented:
Derrell;
At this point I'm not sure these can be merged.  We may be too divergent.

derrell commented:
> Derrell; At this point I'm not sure these can be merged. We may be too divergent.

That may be true, but there has been nothing recently that has been divergent. The two main areas of non-backward compatibility are (1) configuration is in a single configuration file, rather than in one per utility; and (2) group management. My group management currently does not support the subdomains used by NHWG, but could be updated to provide that functionality if you decide you're still interested in merging. (The recent ExpiredV2.py is not a BC issue, as Expired.py remains as it was and the choice of which to use is is a crontab decision.)

derrell committed:
"seniors" means type is either "SENIOR" or "FIFTY YEAR"

derrell committed:
add chaplain corps groups

derrell committed:
misc

derrell committed:
add chaplain corps groups

derrell committed:
completed initial JS script to create shared drive folder structure

derrell committed:
add verbose flag to updateGroupsJob.js; limit output in Group.js by default

derrell committed:
Send custom User-Agent string as required, for address-to-latlon conversion

derrell committed:
correct spelling

derrell committed:
Ensure user is signed out before suspending their account

derrell committed:
Allow groups to specify not to exclude inactive members.

This allows a group to include members of other wings or region, such
as the duty position lists, e.g., wing-staff

derrell committed:
Add new groups *-staff which includes members based on Duty Position
